### PR TITLE
Add Axolotl SFT recipe and organize post-training by objective

### PR DESCRIPTION
**Pull request by AlirezaShamsoshoara**
## Summary

This PR adds a production-oriented supervised fine-tuning recipe for Muse
Glimmer 30B using Axolotl 0.19.0.

It also reorganizes `post-training/` by training objective and framework, making
it straightforward to add LLaMA-Factory, Unsloth, and other approaches later:

```text
post-training/
├── sft/
│   └── axolotl/
└── rl/
    └── titanrl/
        └── data-analyst/
```

### Axolotl SFT recipe

The new recipe covers:

- QLoRA
- BF16 LoRA with FSDP2
- Full-parameter BF16 training with FSDP2
- Text-only data using Tulu
- Single-image data using CoSyn Circuit
- True multi-image data using Docmatix
- Deterministic dataset preparation and processor-level token validation
- W&B experiment tracking
- Fixed-ID base-versus-tuned evaluation
- Docmatix page-removal and page-reordering counterfactuals
- Reproducible training-loss, validation-loss, memory, throughput, and quality
  plots
- Concrete JSON and image examples for each modality

### Measured training results

All five controlled training jobs ran on one 8x H100 node:

| Dataset | Method | Validation loss |
|---|---|---:|
| CoSyn | QLoRA | 2.9434 to 0.8655 |
| CoSyn | BF16 LoRA FSDP2 | 2.8341 to 0.8968 |
| CoSyn | Full BF16 FSDP2 | 2.8341 to 0.8717 |
| Tulu | QLoRA | 1.1078 to 0.7973 |
| Docmatix | QLoRA | 1.0974 to 0.7187 |

Matched held-out evaluation improved token F1 for every dataset and method:

| Comparison | Base token F1 | Tuned token F1 |
|---|---:|---:|
| CoSyn NF4 to QLoRA | 0.1099 | 0.4586 |
| CoSyn BF16 to BF16 LoRA FSDP2 | 0.1096 | 0.4430 |
| CoSyn BF16 to Full BF16 FSDP2 | 0.1096 | 0.4251 |
| Tulu NF4 to QLoRA | 0.3391 | 0.5379 |
| Docmatix NF4 to QLoRA | 0.3926 | 0.6532 |

The Docmatix counterfactuals also produced lower scores when the first page was
removed or page order was reversed, providing evidence that the multi-image path
uses page content and order.

### Muse-specific findings

The recipe documents two important compatibility findings:

- QLoRA reload requires a BF16 model load dtype and BF16 quantization storage.
- FSDP2 activation checkpointing must remain disabled for BF16 LoRA with
  Axolotl 0.19.0. Otherwise checkpoint-wrapper names leak into the gathered
  adapter and PEFT skips the trained weights during reload.

### Post-training hierarchy

The existing TitanRL data-analyst recipe moves under
`post-training/rl/titanrl/`. Its setup now explicitly copies, registers, and
tests the recipe module before launch.

The new hierarchy separates training objectives from framework implementations:

- `post-training/sft/` compares SFT frameworks.
- `post-training/rl/` compares RL frameworks.
- Each framework owns its configs, scripts, tests, assets, and measured results.

## Validation

- 46 Axolotl recipe tests passed
- 17 TitanRL data-analyst tests passed
- Ruff passed
- All three Axolotl configurations parsed from the new recipe directory
- Local Markdown links and embedded images were validated
- Documented Axolotl entry points resolve from the new directory
- TitanRL installation and registration commands were tested in an isolated
  checkout skeleton
- Plot generation is reproducible from the checked-in W&B history exports
- Independent review found no blocker or high-severity issue

image

image

## Scope

The measured runs use controlled, audited subsets:

- CoSyn: 2,048 train and 256 validation examples
- Tulu: 2,048 train and 256 validation examples
- Docmatix: 256 train and 36 validation documents

These results demonstrate that the training paths execute successfully and show
directional held-out improvement. They are not full-dataset production
benchmarks.

### CODE REVIEW

**Review by sid-rp — APPROVED** [review]
LG
